File: ssr_pkg/scripts/chao_node.py
#!/usr/bin/env python3
#coding=utf-8

import rospy
from std_msgs.msg import String
from qq_msgs.msg import Carry

if __name__ == '__main__':
    try:
        # 初始化节点，命名为"chao_node"
        rospy.init_node('chao_node', anonymous=False)
        
        # 创建发布者，发布String类型消息到chao_topic话题，队列长度10
        pub = rospy.Publisher('chao_topic', Carry, queue_size=10)
        
        # 设置发布频率为1Hz
        rate = rospy.Rate(1)  # 1 Hz
        
        # 创建消息对象
        msg = Carry()
        msg.grade = "王者" 
        msg.star = 50
        msg.data = "test_chao"
        
        # 循环发布消息
        while not rospy.is_shutdown():
            # 发布消息
            pub.publish(msg)
            
            # 打印日志信息
            rospy.loginfo("Publishing: %s", msg.data)
            
            # 按照设定频率休眠
            rate.sleep()
            
    except rospy.ROSInterruptException:
        pass

File: ssr_pkg/src/chao_node.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>
#include <qq_msgs/Carry.h>


int main(int argc, char *argv[])
{
    // 设置环境变量，强制使用UTF-8编码
    setenv("LC_ALL", "en_US.UTF-8", 1);

    ros::init(argc, argv, "chao_node");
    ros::NodeHandle nh;
    
    // 创建一个发布者，发布std_msgs/String类型的消息到chao_topic话题，队列长度10
    ros::Publisher pub = nh.advertise<qq_msgs::Carry>("chao_topic", 10);
    
    // 设置发布频率为1Hz
    ros::Rate rate(1);
    
    qq_msgs::Carry msg;
    msg.grade = "王者";
    msg.star = 1;
    msg.data = "test_chao";
    
    while (ros::ok())
    {
        // 发布消息
        pub.publish(msg);
        ROS_INFO("Publishing: %s", msg.data.c_str());
        // printf("刷屏\n");

        // 处理回调函数
        ros::spinOnce();
        
        // 按照设定频率休眠
        rate.sleep();
    }
    
    return 0;
}